flocker/node/agents/cinder.py

```diff
@@ -311,28 +312,25 @@ def reached_desired_state(self):
         Raise an exception if a non-valid state is reached or if the
         desired state is not reached within the supplied time limit.
         """
-        # Could be more efficient.  FLOC-1831
-        for listed_volume in self.volume_manager.list():
-            if listed_volume.id == self.expected_volume.id:
-                # Could miss the expected status because race conditions.
-                # FLOC-1832
-                current_state = listed_volume.status
-                if current_state == self.desired_state:
-                    return listed_volume
-                elif current_state in self.transient_states:
-                    # Once an intermediate state is reached, the prior
-                    # states become invalid.
-                    idx = self.transient_states.index(current_state)
-                    if idx > 0:
-                        self.transient_states = self.transient_states[idx:]
-                else:
-                    raise UnexpectedStateException(
-                        self.expected_volume, self.desired_state,
-                        current_state)
-        elapsed_time = time.time() - self.start_time
-        if elapsed_time > self.time_limit:
-            raise TimeoutException(
-                self.expected_volume, self.desired_state, elapsed_time)
+        try:
+            existing_volume = self.volume_manager.get(self.expected_volume.id)
+            current_state = existing_volume.status
+            if current_state == self.desired_state:
+                return existing_volume
+            elif current_state in self.transient_states:
+                # Once an intermediate state is reached, the prior
+                # states become invalid.
+                idx = self.transient_states.index(current_state)
+                if idx > 0:
+                    self.transient_states = self.transient_states[idx:]
+            else:
+                raise UnexpectedStateException(
+                    self.expected_volume, self.desired_state, current_state)
+        except CinderClientNotFound:
```

Comment:
This exception handling is just for the `get` call so it should only be applied to the `get` call.  Generally, try to keep the `try` block as short as possible.